pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/SchemasResourceBase.java

```diff
@@ -123,8 +125,41 @@ public CompletableFuture<SchemaVersion> deleteSchemaAsync(boolean authoritative,
                 });
     }
 
+    protected CompletableFuture<Void> checkSchemaTypeSupported(PostSchemaPayload payload) {
+        switch (SchemaType.valueOf(payload.getType())) {
+            case AVRO : {
+                Schema schema = SchemaUtil.parseAvroSchema(payload.getSchema());
+                switch (schema.getType()) {
+                    case RECORD: {
+                        break;
+                    }
+                    case UNION: {
+                        return CompletableFuture.failedFuture(new RestException(
+                                Response.Status.BAD_REQUEST.getStatusCode(),
+                                "[" + String.valueOf(topicName) + "] Avro schema typed [UNION] is not supported"));
+                    }
+                    case INT:
+                    case LONG:
+                    case FLOAT:
+                    case DOUBLE:
+                    case BOOLEAN:
+                    case STRING:
+                    case BYTES: {
```

Comment:
would it make sense to have a default case instead?

Reply:
> would it make sense to have a default case instead?

No, if users can register an Avro schema successfully, they will try to use the same schema to consume, and then they will get an error. We'd better deny this operation

Reply:
> > would it make sense to have a default case instead?
> 
> No, if users can register an Avro schema successfully, they will try to use the same schema to consume, and then they will get an error. We'd better deny this operation

@poorbarcode what I meant is using a `default` case for the `switch`, instead of listing all other cases. Does your comment apply to that?

Reply:
@poorbarcode Thanks for checking. Yes, the current approach makes sense. I'll mark this resolved.